assets/app/components/site/editor/configs/pageSettings.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PageSettingsForm from './pageSettingsForm';
+import Codemirror from './codemirror';
+import { parseYaml, writeYaml } from '../../../../util/parseYaml';
+
+const propTypes = {
+  frontmatter: React.PropTypes.string.isRequired,
+  onChange: React.PropTypes.func.isRequired,
+  templateConfig: React.PropTypes.string.isRequired
+};
+
+const getPageLayoutsFromConfig = (config) => {
+  if (!config.defaults) {
+    return [];
+  }
+
+  return config.defaults[0].values.layout;
+};
+
+const buildBaseFormFields = (layouts, handleChange) => {
```

Comment:
Not super happy about having this logic here, its making the component difficult to test. Not sure what the right abstraction is for this collection of private methods